lib/SPIRV/SPIRVRegularizeLLVM.cpp

```diff
@@ -563,6 +582,19 @@ bool SPIRVRegularizeLLVMBase::regularize() {
           }
         }
 
+        if (II.isLogicalShift() &&
+            II.getOperand(0)->getType()->isIntegerTy(1)) {
+          if (II.getOpcode() == Instruction::LShr) {
```

Comment:
`isLogicalShift()` returns true for right-shift `shr` instruction too, which is mapped to `OpShiftLeftLogical` SPIR-V operation. Should we do the same extension for this instruction too?